### PR TITLE
🔥 feat: Add support for creating Fiber client from existing FastHTTP client

### LINKED ISSUE #3107: 📝 [Proposal]: Add support to create client from existing client
### Feature Proposal Description

I believe there should be a way to pass custom fasthttp client in to the fiber http client, to allow more granular fine tuning by the users. 

### Alignment with Express API

---

### HTTP RFC Standards Compliance

---

### API Stability

---

### Feature Examples

```go
Naming can be improved but something along the lines:

cc := client.NewWithFasthttpClien(&fasthttp.Client{...})
```
```


### Checklist:

- [X] I agree to follow Fiber's [Code of Conduct](https://github.com/gofiber/fiber/blob/master/.github/CODE_OF_CONDUCT.md).
- [X] I have searched for existing issues that describe my proposal before opening this one.
- [X] I understand that a proposal that does not meet these guidelines may be closed without explanation.

### PR DESCRIPTION
**Pull request by mitulagr2**
# Description

Add support for passing a custom fasthttp client to the fiber http client, to allow more granular fine tuning by the users.

Fixes #3107 

## Type of change

- [x] New feature (non-breaking change which adds functionality)
  - Add `NewWithClient` method to client.
- [x] Documentation update (changes to documentation)
  - Add method to [client docs](https://github.com/gofiber/fiber/blob/main/docs/client/rest.md#usage).
- [x] Code consistency (non-breaking change which improves code reliability and robustness)
  - The `New` method calls `NewWithClient` internally.
  - Add unit tests for valid fasthttp client and nil client.

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -680,8 +680,13 @@ func New() *Client {
 	// trie to use a pool to reduce the cost of memory allocation
 	// for the fiber client and the fasthttp client
 	// if possible also for other structs -> request header, cookie, query param, path param...
+	return NewWithClient(&fasthttp.Client{})
+}
+
+// NewWithClient creates and returns a new Client object from an eisting client object.
```
Comment (gaby) [inline client/client.go:686]: Typo "existing"

File: client/client.go [hunk 2]
```diff
@@ -680,8 +680,13 @@ func New() *Client {
...
+func NewWithClient(c *fasthttp.Client) *Client {
```
Comment (gaby) [inline client/client.go:687]: We need a unit-test for this function.

**Review by gaby — CHANGES_REQUESTED** [review]

File: client/client.go (hunk 1 above)
Comment (gaby) [reply client/client.go:686]: Remove redundant "object" at the end of sentence.

File: client/client.go (hunk 2 above)
Comment (mitulagr2) [reply client/client.go:687]: On it!

File: client/client.go [hunk 3]
```diff
@@ -680,8 +680,13 @@ func New() *Client {
 	// trie to use a pool to reduce the cost of memory allocation
 	// for the fiber client and the fasthttp client
 	// if possible also for other structs -> request header, cookie, query param, path param...
+	return NewWithClient(&fasthttp.Client{})
+}
+
+// NewWithClient creates and returns a new Client object from an existing client.
+func NewWithClient(c *fasthttp.Client) *Client {
```
[the remark this answers is not in the document]
Comment (gaby) [reply client/client.go:687]: @mitulagr2 Can you add the nil check suggested including the panic.

**Review by gaby — CHANGES_REQUESTED** [review]

File: client/client.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (gaby) [reply client/client.go:687]: Also add a test where you call the function with `nil`, and test that it panics. We use testify you can check this with:

```go
require.Panics(t, func(){
    // create client with nil here to throw panic
})
```

**Review by gaby — APPROVED** [review]
👍 LGTM

**Review by efectn — APPROVED** [review]

**Review by ReneWerner87 — APPROVED** [review]
